### PR TITLE
NETOBSERV-789 UI: Table Histogram shortcuts

### PR DESCRIPTION
**Pull request by jpinsonneau**
This improves histogram user experience by adding buttons and keyboards shortcuts for the following actions:
- move range previous / next
  - binded to left and right arrow keys
- zoom / unzoom range
  - binded to minus and plus buttons
- move histogram selection previous / next
  - binded to page up and down

image

To let the user better understand what's happening, the custom time range label has been replaced by actual values.

image

Also I suggest to decrease default limit to 50 as queries will be faster and this new method allows a better navigation. The best option for me would be to replace the current selector by a [number input](https://www.patternfly.org/v4/components/number-input#default) in a followup

image

based on https://github.com/netobserv/network-observability-console-plugin/pull/271

### CODE REVIEW

**Comment by jpinsonneau** [comment]
@jotak @OlivierCazade I have merged https://github.com/netobserv/network-observability-console-plugin/pull/271 and rebased this PR so you can review these extra changes when you have some time

@andrew-ronaldson feel free to give your feedback here or in a followup

Thank you !

**Comment by Amoghrd** [comment]
So @jpinsonneau The `<>` arrows moves the datapoints by 2 mins; `<<>>` arrows move the time range by 15 minutes and `+-` moves the time range by 30 minutes. thats the expected behavior right?
But one nit, the custom time range field can only take values which are given in the dropdown(which is every 30 mins), something like 14:22:23 cannot be one given as inputs.

**Comment by jpinsonneau** [comment]
> So @jpinsonneau The `<>` arrows moves the datapoints by 2 mins; `<<>>` arrows move the time range by 15 minutes and `+-` moves the time range by 30 minutes. thats the expected behavior right? 

Well these values are not fixed; it's context related:
-  `<>` shift the selection inside the histogram and try to avoid limit reached by adapting the new width. If it can't, it will keep the previous selection width.
- `<<>>`  move the time range by currently selected range on top right corner time range dropdown (Last 15 minutes in your example)
- `+-`  select previous / next item in the list of the top right corner time range dropdown

> But one nit, the custom time range field can only take values which are given in the dropdown(which is every 30 mins), something like 14:22:23 cannot be one given as inputs.

I'm not sure to get that point. Custom time range is fully free. You can pick any value there and the histogram shortcuts will do so.

**Review by OlivierCazade — APPROVED** [review]
LGTM, my comment was more a general comment

File: web/src/components/netflow-traffic.tsx [hunk 1]
```diff
@@ -1005,6 +1007,70 @@ export const NetflowTraffic: React.FC<{
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [match, filters]);
 
+  const moveRange = React.useCallback(
```
[the remark this answers is not in the document]
Comment (jpinsonneau) [reply web/src/components/netflow-traffic.tsx:1010]: Fully aligned with that. I'm planning to [move the drawer away](https://github.com/netobserv/network-observability-console-plugin/commit/0458519301ba2cb2fcf7565bb424b1bf800e34f0#diff-6025777eb7820d14f17449d64e9adb6086d9b2c343e057329428892ecb28ef36) as I did in my PoC on connection tracking.

We can also move the modals away from that file creating a dedicated component.

I'm adding a note in https://issues.redhat.com/browse/NETOBSERV-268 to keep track.

**Pull request merged**